sycl/test-e2e/ExtendedDeleters/extended_deleters.cpp

```diff
@@ -0,0 +1,18 @@
+// REQUIRES: hip, cuda
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out | FileCheck %s
+
+// CHECK: This extended deleter should be called at ctx destruction.
+
+#include <sycl/sycl.hpp>
+
+int main() {
+  sycl::context c;
+  sycl::detail::pi::contextSetExtendedDeleter(
+      c,
+      [](void *) {
+        printf("This extended deleter should be called at ctx destruction.");
+      },
+      nullptr);
+}
```

Comment:
Why do we need it here and not in the UR repo? This makes no sense to me... `sycl::detail::pi` has to be eradicated from the sycl repo, IMO.

Reply:
If the test does move to UR it'll need to be in a separate PR.

Reply:
We already have a test for this in the UR repo but this is something that is used in oneMKL through the `sycl::detail::pi` interface. I think it's worthwhile having a test to make sure that the SYCL interface is working as well as the UR one.

Reply:
> something that is used in oneMKL through the sycl::detail::pi interface.

That is bad... {{detail}} shouldn't be used in production.

Reply:
I agree but I think that ship has sailed. See here https://github.com/oneapi-src/oneMKL/blob/develop/src/blas/backends/cublas/cublas_scope_handle.cpp#L123

Reply:
There have been some changes in UR contexts of late, and when these are settled we are going to re-review this entry point and see if we can potentially remove it from oneMKL, but as it stands it is necessary

Reply:
I've submitted oneapi-src/oneMKL#401. Personally, I'm in favor of removing the test: we do not guarantee availability of the tested functionality. The fact that it resides in `sycl::detail` namespace explicitly communicates that.

Reply:
I think we should remove this test whenever we stop using this in oneMKL. The reason why I am adding this test is because this functionality was accidentally removed, and since there was no testing for it we didn't realise that something was broken, and oneMKL ended up breaking.